game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java

```diff
@@ -112,9 +113,9 @@ private void addChat() {
     remove(mainPanel);
     remove(chatSplit);
     chatPanelHolder.removeAll();
-    final ChatModel chat = chatPanelSupplier.get().orElse(null);
-    if (chat instanceof SetupPanel) {
-      chatPanelHolder.add((SetupPanel) chat, BorderLayout.CENTER);
+    final ChatModel chat = chatModelSupplier.get().orElse(null);
+    if ((chat != null) && !chat.isHeadless()) {
+      chatPanelHolder.add(new ChatPanel(chat.getChat()), BorderLayout.CENTER);
```

Comment:
I just realized what is the problem here.
Your change is overkill here, I opened #4788 to fix this issue

Reply:
Using an instanceof was considered, per overview:

> One fix considered was to update the instanceof check to check for presumably the right chat model type 'HeadlessChatModel', but this still would have left in the brittle instanceof check. The fix here rolls back to an explicit 'isHeadless' method that was previously used to do this check.

If use instanceof and change the `ChatPanel` type to be a new subclass, we'll have to change 'n' locations. (and that is *if* we notice and do not get back into a situation where we silently and unknowingly break a major feature. At least this was a noticeable problem and fixed before release)

Reply:
@RoiEXLab even if #4788 were merged, I would still submit this PR and then consider a tech debt cleanup instead of a bug fix.